app/components/instruction/ed25519/Ed25519DetailsCard.tsx

```diff
@@ -0,0 +1,162 @@
+import {
+    ParsedTransaction,
+    PartiallyDecodedInstruction,
+    PublicKey,
+    SignatureResult,
+    TransactionInstruction,
+} from '@solana/web3.js';
+import bs58 from 'bs58';
+import React from 'react';
+
+import { Address } from '../../common/Address';
+import { Copyable } from '../../common/Copyable';
+import { InstructionCard } from '../InstructionCard';
+import { PROGRAM_ID as ED25519_PROGRAM_ID } from './types';
+
+type DetailsProps = {
+    tx: ParsedTransaction;
+    ix: TransactionInstruction;
+    index: number;
+    result: SignatureResult;
+    innerCards?: JSX.Element[];
+    childIndex?: number;
+};
+
+interface Ed25519SignatureOffsets {
+    signatureOffset: number; // offset to ed25519 signature of 64 bytes
+    signatureInstructionIndex: number; // instruction index to find signature
+    publicKeyOffset: number; // offset to public key of 32 bytes
+    publicKeyInstructionIndex: number; // instruction index to find public key
+    messageDataOffset: number; // offset to start of message data
+    messageDataSize: number; // size of message data
+    messageInstructionIndex: number; // index of instruction data to get message data
+}
+
+// See https://docs.anza.xyz/runtime/programs/#ed25519-program
+function decodeEd25519Instruction(data: Buffer): { count: number; offsets: Ed25519SignatureOffsets[] } {
+    const count = data.readUInt8(0);
+    const offsets: Ed25519SignatureOffsets[] = [];
+
+    let cursor = 2; // Skip count and padding byte
+
+    for (let i = 0; i < count; i++) {
+        const offset: Ed25519SignatureOffsets = {
+            messageDataOffset: data.readUInt16LE(cursor + 8),
+            messageDataSize: data.readUInt16LE(cursor + 10),
+            messageInstructionIndex: data.readUInt16LE(cursor + 12),
+            publicKeyInstructionIndex: data.readUInt16LE(cursor + 6),
+            publicKeyOffset: data.readUInt16LE(cursor + 4),
+            signatureInstructionIndex: data.readUInt16LE(cursor + 2),
+            signatureOffset: data.readUInt16LE(cursor),
+        };
+        offsets.push(offset);
+        cursor += 14; // Size of one Ed25519SignatureOffsets struct
+    }
+
+    return { count, offsets };
```

Comment:
returned count is not needed/used. You can remove it unless there's another usage you're expecting